homework/index.js

```diff
@@ -30,13 +30,103 @@
     return elem;
   }
 
+  function removeChildren(element) {
+    while (element.firstChild) {
+      element.removeChild(element.firstChild);
+    }
+  }
+
   function main(url) {
     fetchJSON(url, (err, data) => {
       const root = document.getElementById('root');
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        const jsonString = JSON.stringify(data, null, 2);
+        const menu = createAndAppend('div', root, {
+          id: 'menuSelect',
+        });
+        createAndAppend('h2', menu, { text: 'HYF Repositories' });
+        const select = createAndAppend('select', menu, {
+          id: 'selection',
+        });
+        const repoDetails = createAndAppend('div', root, {
+          id: 'repoDetails',
+        });
+        createAndAppend('p', repoDetails, {
+          id: 'repoDescriptionP',
+        });
+        createAndAppend('p', repoDetails, {
+          id: 'forks',
+        });
+
+        createAndAppend('p', repoDetails, {
+          id: 'updateDate',
+        });
+
+        const renewalSpace = createAndAppend('div', root, {
+          id: 'contributors',
+        });
+
+        let valueNumber = 0;
+        const repositories = JSON.parse(jsonString, null);
+        let contributorsURL = '';
+
+        repositories.forEach(repo => {
```

Comment:
1. The assignment asked for the repositories to be sorted by name. 
2. Instead of using a separate `valueNumber` variable you can use the optional `index` variable from `forEach`:

   ```js
   repositories.forEach((repo, index) => {
     ...
     value: index,
     ...
   });
   ```

Reply:
Done!

Reply:
Now, its sorting repositories by the name. Also figured it out the problem of showing first thing when the page loaded in week2's homework.